UIforETW/UIforETWDlg.cpp

```diff
@@ -1646,7 +1646,7 @@ void CUIforETWDlg::PreprocessTrace(const std::wstring& traceFilename)
 		child.Run(bShowCommands_, L"xperf.exe" + args);
 		std::wstring output = child.GetOutput();
 		std::map<std::wstring, std::vector<DWORD>> pidsByType;
-		for (auto& line : split(output, '\n'))
+		for (const auto line : split(output, '\n'))

```

Comment:
No need for a temporary variable. The ideal code is:

    for (const auto& line : split(output, '\n'))

Adding a 'lines' local variable is unnecessary, and would make the result of the split() function live longer than necessary. I think you're confusing the result of split() and the 'line' variable. The compiler is required to hang on to the return value of the split() function until containing expression (the for loop) completes. If it doesn't then it doesn't matter what the type of the 'line' variable is because it would be iterating through a destroyed vector. Therefore 'line' should be a reference because that is more efficient.

That is, making 'line' not be a reference wouldn't solve the problem you are concerned with, and it isn't a problem anyway.

I also like that the line of code ends up looking pythonic when we iterate over the return value directly. When C++11 lets code look more like Python I think that is a good thing.